### DISCUSSION TITLE
Is it possible to encrypt SFS Capacity-Oriented volume via terraform?

### DISCUSSION THREAD

**Discussion by tobias-dehn-apoplex** [opentelekomcloud/terraform-provider-opentelekomcloud#1750] [Q&A]
Hi all,

I'm trying to create an SFS Capacity-Oriented volume via terraform. I'm wondering why I cannot encrypt a opentelekomcloud_sfs_file_system_v2 resource but a opentelekomcloud_sfs_turbo_share_v1?

Although it is possible via the OTC console.

Best regards
Tobi

**Comment by lego963** [comment]
@tobias-dehn-apoplex hiho,
Answer from Storage Squad:
```
SFS file system is deprecated service. It doesn't work from console, only API. Try migrating to SFS Turbo if it's possible.
```

**Reply by tobias-dehn-apoplex** [reply]
@lego963 Hi,

Thank you for your fast response. It is only deprecated for the OTC terraform provider, or even for OTC? Where can I look up this information?

I found a workaround utilizing the [API](https://docs.otc.t-systems.com/en-us/api/sfs/sfs_02_0021.html). If I add the corresponding values to metadata, I can achieve encryption enabled.
```
resource "opentelekomcloud_sfs_file_system_v2" "sfs_1" {
  name        = "test"
  size        = 50
  share_proto = "NFS"
  is_public = false
  metadata = {
    "#sfs_crypt_key_id": data.opentelekomcloud_kms_key_v1.test_key.id,
    "#sfs_crypt_alias": data.opentelekomcloud_kms_key_v1.test_key.key_alias,
    "#sfs_crypt_domain_id": data.opentelekomcloud_kms_key_v1.test_key.domain_id
  }
}
```

**Reply by lego963** [reply]
@tobias-dehn-apoplex we will deprecate it in tf provider. It's deprecated for the OTC. 
https://open-telekom-cloud.com/en/support/release-notes/scalable-file-service/sfs-sold-out-but-sfs-turbo-open
